test/unit/queue/src/CollectiveQueue.cpp

```diff
@@ -106,9 +106,8 @@ TEST_CASE("TestCollectiveMemcpy", "[queue]")
         auto dst = alpaka::createView(dev, results.data() + threadId, Vec(static_cast<Idx>(1u)), Vec(sizeof(int)));
         auto src = alpaka::createView(dev, &threadId, Vec(static_cast<Idx>(1u)), Vec(sizeof(int)));
 
-        // avoid that the first thread is executing the copy (can not be guaranteed)
-        size_t sleep_ms = (results.size() - static_cast<uint32_t>(threadId)) * 100u;
-        std::this_thread::sleep_for(std::chrono::milliseconds(sleep_ms));
+        // avoid that the first thread is executing the memcopy before all other threads wrote there changes
+        alpaka::wait(queue);
```

Comment:
I don't understand why the wait is necessary. The sleep before is optional, only to try to make another thread as the main thread run the memcpy.

Reply:
Wait will guarantee that all threads reached the point after line 107. The old sleep was hoping that we have no data-race by waiting long enouph.

Reply:
But this is not necessary, the following program is fine too:
```c++
#    pragma omp parallel num_threads(static_cast <int>(results.size()))
    {
        int threadId = omp_get_thread_num();
        auto dst = alpaka::createView(dev, results.data() + threadId, Vec(static_cast<Idx>(1u)), Vec(sizeof(int)));
        auto src = alpaka::createView(dev, &threadId, Vec(static_cast<Idx>(1u)), Vec(sizeof(int)));

        // no sleep and wait

        // only one thread will perform this memcpy
        alpaka::memcpy(queue, dst, src, Vec(static_cast<Idx>(1u)));
        alpaka::wait(queue);
    }
```
Where do you spot a data race that you are trying to prevent?

Reply:
You are right and it was most likely a false positive in the thread sanitizer.
I will remove the change from this PR and update the documentation of the COllectiveQueue only.